src/main/filters/SpectralTilt.cpp

```diff
@@ -0,0 +1,390 @@
+            }
+
+            if ((enSlopeUnit == STLT_SLOPE_UNIT_NONE) || (fSlopeNepNep == 0.0f))
+            {
+                bBypass = true;
+                bSync = true;
+                return;
+            }
+            else
+            {
+                bBypass = false;
+            }
+
+            float l_angf = 2.0f * M_PI * fLowerFrequency;
+            float u_angf = 2.0f * M_PI * fUpperFrequency;
+
+            // Exponential spacing ratio for poles.
+            float r = powf(u_angf / l_angf, 1.0f / (nOrder - 1));
+            float c_fn = bilinear_coefficient(1.0f, nSampleRate);
+            float negZero = l_angf * powf(r, -fSlopeNepNep);
+            float negPole = l_angf;
+
+            // We have nOrder bilinears. We combine them 2 by 2 to get biquads.
+
+            sFilter.begin();
+            for (size_t n = 0; n < nOrder; ++n)
+            {
+                if (n % 2 != 0)
+                    continue;
+
+                bilinear_spec_t spec_now = compute_bilinear_element(negZero, negPole);
+                negZero *= r;
+                negPole *= r;
+
+                bilinear_spec_t spec_nxt = compute_bilinear_element(negZero, negPole);
+                negZero *= r;
+                negPole *= r;
+
+                dsp::biquad_x1_t *digitalbq = sFilter.add_chain();
+                if (digitalbq == NULL)
+                    return;
+
+                dsp::f_cascade_t analogbq;
+                analogbq.t[0] = spec_now.b0 * spec_nxt.b0;
+                analogbq.t[1] = spec_now.b0 * spec_nxt.b1 + spec_now.b1 * spec_nxt.b0;
+                analogbq.t[2] = spec_now.b1 * spec_nxt.b1;
+                analogbq.b[0] = spec_now.a0 * spec_nxt.a0;
+                analogbq.b[1] = spec_now.a0 * spec_nxt.a1 + spec_now.a1 * spec_nxt.a0;
+                analogbq.b[2] = spec_now.a1 * spec_nxt.a1;
+
+                dsp::bilinear_transform_x1(digitalbq, &analogbq, c_fn, 1);
+
+                // The denominator coefficients in digitalbq will have opposite sign with respect the maths.
+                // This is correct, as this is the LSP convention.
+            }
+            sFilter.end(true);
+
+            bSync = true;
+        }
+
+        void SpectralTilt::process_add(float *dst, const float *src, size_t count)
+        {
+            if (src != NULL)
+                dsp::copy(dst, src, count);
+            else
+                dsp::fill_zero(dst, count);
+
+            if (bBypass)
+            {
+                dsp::mul_k2(dst, 2.0f, count);
+                return;
+            }
+
+            while (count > 0)
+            {
+                size_t to_do = lsp_min(count, BUF_LIM_SIZE);
+
+                sFilter.process(vBuffer, dst, to_do);
+                dsp::add2(dst, vBuffer, to_do);
+
+                dst     += to_do;
+                count   -= to_do;
+            }
+
+        }
+
+        void SpectralTilt::process_mul(float *dst, const float *src, size_t count)
```

Comment:
I think this code can be refactored to:
```C++
void SpectralTilt::process_mul(float *dst, const float *src, size_t count)
{
    float buffer[BUF_LIM_SIZE];

    if (src == NULL)
    {
        // No inputs, interpret `src` as zeros: dst[i] = dst[i] * 0 = 0
        dsp::fill_zero(dst, count);
        return;
    }
    else if (bBypass)
    {
        // Bypass is set: dst[i] = dst[i] * src[i]
        dsp::mul2(dst, src, count);
        return;
    }
    
    for (size_t offset=0; offset < count; )
    {
        size_t to_do = lsp_min(count - offset, BUF_LIM_SIZE);

        // dst[i] = dst[i] * filter(src[i])
        sFilter.process(buffer, &src[offset], to_do);
        dsp::mul2(&dst[offset], buffer, to_do);

        offset += to_do;
     }
}
```

This will work more correct and in most optimal way I think